packages/babel-plugin/src/utils/ast.tsx

```diff
@@ -260,7 +260,9 @@ export const resolveBindingNode = (
   binding: Binding | undefined,
   meta: Metadata
 ): PartialBindingWithMeta | undefined => {
-  if (!binding) {
+  if (!binding || binding.path.isObjectPattern()) {
```

Comment:
🤔 Can there be more use cases like this? Just thinking 🤔 If yes, then this **or** condition will go up 😞

Reply:
most likely tbh - but im sure its a finite list

Reply:
> most likely tbh - but im sure its a finite list
> 
> can we think of any other scenarios that we've missed (when it comes to bindings referencing something in scope)? i can't think of any.

I just pulled the branch. There are 2 things:

1. It's still throwing "Maximum call stack size exceeded" error for this use case 😞
```typescript
import '@compiled/core';

export const DestructuredComp = (props) => {
  const { foo, color } = props;

  return (
    <>
      <span css={{ fontSize: foo, color, backgroundColor: 'blue' }} />
    </>
  );
};
```
2. We are skipping the resolving part right, but what if its coming from a variable outside the function 🤔? May be from a variable declarator or import. Should we also skip in that case 🤔? Something like this:

```typescript
import '@compiled/core';

const { foo } = { foo: 10 };

export const DestructuredComp = ({ color }) => {
  return (
    <>
      <span css={{ fontSize: foo, color, backgroundColor: 'blue' }} />
    </>
  );
};
```

Reply:
awesome insight man, i'll do some testing this morning and fix anything that breaks. we should still be supporting proper de-structuring - perhaps we don't right now?

good catch for the maximum call stack exceed - i'll write a text